### PR TITLE
Typosquatting: finish configuration file and feature flags

### PR DESCRIPTION
**Pull request by zhhyu**
Finish the typo-squatting configuration file and feature flags. 

Fix: https://github.com/NuGet/Engineering/issues/1596

### CODE REVIEW

File: src/NuGetGallery/Services/TyposquattingConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace NuGetGallery.Services
+{
+    public sealed class TyposquattingConfiguration : ITyposquattingConfiguration
+    {
+        public int ChecklistLength { get; }
+        public bool IsCheckEnabled { get; }
+        public bool IsBlockUsersEnabled { get; }
+        public TyposquattingConfiguration()
+            : this(checklistLength: 20000,
```
Comment (shishirx34) [inline src/NuGetGallery/Services/TyposquattingConfiguration.cs]: set the value as a const field.

File: src/NuGetGallery/Services/ITyposquattingConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGetGallery.Services
+{
+    public interface ITyposquattingConfiguration
+    {
+        int ChecklistLength { get; }
```
Comment (shishirx34) [inline src/NuGetGallery/Services/ITyposquattingConfiguration.cs]: may be rename this to `IdChecklistLength` to be specific?

File: src/NuGetGallery/App_Start/OwinStartup.cs [hunk 3]
```diff
@@ -63,14 +63,18 @@ public static void Configuration(IAppBuilder app)
 
             // Configure machine key for session persistence across slots
             SessionPersistence.Setup(config);
-
+
```
Comment (shishirx34) [inline src/NuGetGallery/App_Start/OwinStartup.cs]: nit: extra spacings.

File: src/NuGetGallery/Services/TyposquattingConfiguration.cs (hunk 1 above)
Comment (zhhyu) [reply src/NuGetGallery/Services/TyposquattingConfiguration.cs]: Done

File: src/NuGetGallery/Services/ITyposquattingConfiguration.cs (hunk 2 above)
Comment (zhhyu) [reply src/NuGetGallery/Services/ITyposquattingConfiguration.cs]: Changed to PackageIdChecklistLength.

File: src/NuGetGallery/App_Start/OwinStartup.cs (hunk 3 above)
Comment (zhhyu) [reply src/NuGetGallery/App_Start/OwinStartup.cs]: Delete it.

**Pull request merged**